site/public/js/grade-inquiry.js

```diff
@@ -80,8 +78,98 @@ function onGradeInquirySubmitClicked(button) {
     type: "POST",
     url: button_clicked.attr("formaction"),
     data: form.serialize(),
-    success: function(){
+    success: function(response){
+      try {
+        let json = JSON.parse(response);
+        if (json['status'] === 'success') {
+          let data = json['data'];
+          let course = document.body.dataset.courseUrl.split('/').pop();
+          let submitter_id = form.children('#submitter_id').val();
+          if (data.type === 'new_post') {
+            let gc_id = form.children('#gc_id').val();
+            window.socketClient.send({
+              'type': data.type,
+              'post_id': data.post_id,
+              'course': course,
+              'submitter_id': submitter_id,
+              'gc_id': gc_id
+            });
+          }
+          else if (data.type === 'toggle_status')
+            window.socketClient.send({'type': data.type, 'course': course, 'submitter_id': submitter_id});
+        }
+      }
+      catch (e) {
+        console.log(e);
+      }
       window.location.reload();
     }
   });
 }
+
+function initGradingInquirySocketClient() {
+  window.socketClient = new WebSocketClient();
+  window.socketClient.onmessage = (msg) => {
+    switch (msg.type) {
+      case "new_post":
+        gradeInquiryNewPostHandler(msg.submitter_id, msg.post_id, msg.gc_id);
+        break;
+      case "toggle_status":
+        gradeInquiryDiscussionHandler(msg.submitter_id);
+        break;
+      default:
+        console.log("Undefined message recieved.");
+    }
+  };
+  window.socketClient.open();
+}
+
+function gradeInquiryNewPostHandler(submitter_id, post_id, gc_id) {
+  $.ajax({
+    type: "POST",
+    url: buildCourseUrl(['gradeable', window.location.pathname.split("gradeable/")[1].split('/')[0], 'grade_inquiry', 'single']),
+    data: {submitter_id: submitter_id, post_id: post_id, csrf_token: window.csrfToken},
+    success: function(new_post){
+      // if grading inquiry per component is allowed
+      if (gc_id != 0){
+        // add new post to all tab
+        let all_inquiries = $(".grade-inquiries").children("[data-component_id='0']");
+        let last_post = all_inquiries.children('.post_box').last();
+        $(new_post).insertAfter(last_post).hide().fadeIn('slow');
+
+        // add to grading component
+        let component_grade_inquiry = $(".grade-inquiries").children("[data-component_id='" + gc_id + "']");
+        last_post = component_grade_inquiry.children('.post_box').last();
+        if (last_post.length == 0) // if no posts
+          last_post = component_grade_inquiry.children('.grade-inquiry-header-div').last();
```

Comment:
Can you add curly braces for the `if` condition even if its single liner condition because its more readable